.github/workflows/test-build-deploy.yml

```diff
@@ -278,91 +282,116 @@ jobs:
       - name: Build Mimir with race-detector
         run: |
           export IMAGE_TAG_RACE=$(make image-tag-race)
-          export MIMIR_IMAGE="grafana/mimir:$IMAGE_TAG_RACE"
+          export MIMIR_IMAGE="grafana/mimir:distroless-$IMAGE_TAG_RACE"
+          export MIMIR_ALPINE_IMAGE="grafana/mimir:$IMAGE_TAG_RACE"
           make BUILD_IN_CONTAINER=false cmd/mimir/.uptodate_race
-          docker save $MIMIR_IMAGE -o ./mimir_race_image
+          docker save $MIMIR_IMAGE -o ./mimir_race_image_distroless
+          docker save $MIMIR_ALPINE_IMAGE -o ./mimir_race_image
       - name: Upload archive with race-enabled Mimir
         uses: actions/upload-artifact@v4
         with:
           name: Race-enabled Mimir
-          path: ./mimir_race_image
+          path: |
+            ./mimir_race_image
+            ./mimir_race_image_distroless
 
   integration:
-    needs: [goversion, build]
+    needs: [goversion, build, prepare]
```

Comment:
(nit) I had a [separate job](https://github.com/grafana/mimir/pull/7428/files#diff-cca4a3e8daf0ba1f4ed7d51f31efa8a142ad6870504fe1425a9e126bbb0a6a18R20) for setting the is_deploy output because using `prepare` here unnecessarily checks out the repository and runs make. It's pretty light weight though so just a nit.